### PR TITLE
DBZ-5267 Invalid date 'SEPTEMBER 31'

### PR DESCRIPTION
**Pull request by harveyyue**
https://issues.redhat.com/browse/DBZ-5267

### CODE REVIEW

**Comment by jpechane** [comment]
@harveyyue Hi, thanks for the report. Could you please take a look at https://debezium.io/documentation/reference/1.9/connectors/mysql.html#mysql-property-event-deserialization-failure-handling-mode and make it compatible/driven by the same config option?

Also does this cover situation when an invalid date is defined as default value?

**Review by jpechane — CHANGES_REQUESTED** [review]
@harveyyue I've left one comment

**Comment by harveyyue** [comment]
> @harveyyue I've left one comment

@jpechane Ok, I will look into your comments.

**Comment by harveyyue** [comment]
@jpechane Done with these review

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RowDeserializers.java [hunk 1]
```diff
@@ -598,4 +639,19 @@ protected static int deserializeFractionalSecondsInNanosNegative(int fsp, ByteAr
     private RowDeserializers() {
     }
 
+    private static Serializable handleException(EventProcessingFailureHandlingMode eventProcessingFailureHandlingMode,
+                                                String columnType, String exception, Serializable defaultValue) {
```
[the remark this answers is not in the document]
Comment (harveyyue) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RowDeserializers.java]: Done.

**Review by jpechane — APPROVED** [review]
@harveyyue That's exactly what we need! Would it be possible to add a test for this too? I am not sure if our test database setup will allow to. If not we will merge it as is.

**Comment by harveyyue** [comment]
> @harveyyue That's exactly what we need! Would it be possible to add a test for this too? I am not sure if our test database setup will allow to. If not we will merge it as is.

If execute following statement in streaming phase, can verify these code changes, but the invalid data in binlog could be read by other IT tests if SnapshotMode is NEVER. So I suggest that don't add the IT test.

`set sql_mode='ALLOW_INVALID_DATES';
insert into DBZ5267(d1, c1) values('2022-09-31', '2022-09-31 09:00:00');`

**Pull request merged**